Libs/vtkITK/vtkITKLabelShapeStatistics.h

```diff
@@ -0,0 +1,103 @@
+/*=========================================================================
+
+  Copyright Brigham and Women's Hospital (BWH) All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+==========================================================================*/
+
+#ifndef __vtkITKLabelShapeStatistics_h
+#define __vtkITKLabelShapeStatistics_h
+
+#include "vtkITK.h"
+#include "vtkSimpleImageToImageFilter.h"
+
+// VTK includes
+#include <vtkMatrix4x4.h>
+#include <vtkSmartPointer.h>
+#include <vtkVector.h>
+
+// std includes
+#include <map>
+
+class vtkPoints;
+
+/// \brief ITK-based utilities for calculating label statistics.
+class VTK_ITK_EXPORT vtkITKLabelShapeStatistics : public vtkSimpleImageToImageFilter
+{
+public:
+  static vtkITKLabelShapeStatistics *New();
+  vtkTypeMacro(vtkITKLabelShapeStatistics, vtkSimpleImageToImageFilter);
+  void PrintSelf(ostream& os, vtkIndent indent) override;
+
+  vtkGetMacro(ComputeFeretDiameter, bool);
+  vtkSetMacro(ComputeFeretDiameter, bool);
+  vtkBooleanMacro(ComputeFeretDiameter, bool);
+
+  vtkGetMacro(ComputeOrientedBoundingBox, bool);
+  vtkSetMacro(ComputeOrientedBoundingBox, bool);
+  vtkBooleanMacro(ComputeOrientedBoundingBox, bool);
+
+  vtkGetMacro(ComputePerimeter, bool);
+  vtkSetMacro(ComputePerimeter, bool);
+  vtkBooleanMacro(ComputePerimeter, bool);
+
+  vtkSetObjectMacro(Directions, vtkMatrix4x4);
+
+public:
+  void ClearCentroids();
+  void AddCentroid(int labelValue, vtkVector3d centroid);
+  bool GetCentroid(int labelValue, double* centroid);
+
+  void ClearOrientedBoundingBox();
+  void AddBoundingBox(int labelValue, vtkMatrix4x4* directions, vtkVector3d origin, vtkVector3d size, vtkPoints* points);
+  void GetOrientedBoundingBoxDirection(int labelValue, vtkMatrix4x4* direction);
+  void GetOrientedBoundingBoxOrigin(int labelValue, double* origin);
+  void GetOrientedBoundingBoxSize(int labelValue, double* size);
+  void GetOrientedBoundingBoxVertices(int labelValue, vtkPoints* points);
+
+  void ClearFeretDiameter();
+  void AddFeretDiameter(int labelValue, double feretDiameter);
+  double GetFeretDiameter(int labelValue);
+
+  void ClearPerimeter();
+  void AddPerimeter(int labelValue, double perimeter);
+  double GetPerimeter(int labelValue);
+
+  void ClearRoundness();
+  void AddRoundness(int labelValue, double roundness);
+  double GetRoundness(int labelValue);
+
+  void ClearFlatness();
```

Comment:
There are a number of other parameters, such as principal axes directions that would be useful to expose. How have you decided what to include?
It would be nice to expose all possible parameters, but as of now, it would be a lot of copy-paste. Maybe we could add macros or allow enabling/disabling computation and getting values via string keys. vtkVariant could be used as return value type.